app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java

```diff
@@ -0,0 +1,240 @@
+        for (int i = 0; i < ret.length; i++) {
+            ret[i] = iterator.next().intValue();
+        }
+        return ret;
+    }
+
+    private TreeMap <Integer, Integer> setUpDailyTextMap() {
+
+        TreeMap<Integer, Integer> weeklyMap = new TreeMap<>();
+
+        weeklyMap.put(SUN, DEFAULT_VALUE);
+        weeklyMap.put(MON, DEFAULT_VALUE);
+        weeklyMap.put(TUE, DEFAULT_VALUE);
+        weeklyMap.put(WED, DEFAULT_VALUE);
+        weeklyMap.put(THUR, DEFAULT_VALUE);
+        weeklyMap.put(FRI, DEFAULT_VALUE);
+        weeklyMap.put(SAT, DEFAULT_VALUE);
+
+        return weeklyMap;
+    }
+
+    private void getHigehstValueForYaxis() {
+
+        highestValue = getYValue(sentValues, receivedValues);
+    }
+
+    synchronized private int getYValue(float[] sentValues, float[] receivedValues) {
+        int maxSent = findMaximumValue(sentValues);
+        int maxReceived = findMaximumValue(receivedValues);
+        if (maxSent > maxReceived) {
+            return highestValue = (int) getRound(maxSent);
+        }
+        return highestValue = (int) getRound(maxReceived);
+    }
+
+    private static int findMaximumValue(float[] inputArray) {
+
+        float maxValue = inputArray[0];
+
+        for (int i = 1; i < inputArray.length; i++) {
+
+            if (inputArray[i] > maxValue) {
+                maxValue = Math.round(inputArray[i]);
+            }
+        }
+        return (int) maxValue;
+    }
+
+    private long getRound(int input) {
+
+        return Math.round(input * 1.25);
+    }
+
+    synchronized private void loadGraph() {
+
+        setGraphData();
+        setGraphAttributes();
+        animateGraph();
+    }
+
+    private void setGraphData() {
+
+        final String[] xAxisLabels = {
+
+                context.getString(R.string.sun),
+                context.getString(R.string.mon),
+                context.getString(R.string.tue),
+                context.getString(R.string.wed),
+                context.getString(R.string.thu),
+                context.getString(R.string.fri),
+                context.getString(R.string.sat),
+        };
+
+        LineSet dataSet = new LineSet(xAxisLabels, receivedValues);
+        dataSet.setColor(parseColor(YELLOW_CRAYOLA))
+                .setDotsColor(parseColor(RED_ROSE_MADDER))
+                .setFill(parseColor(BLUE_SAPPHIRE))
+                .setThickness(6)
+                .beginAt(0);
+        lineGraph.addData(dataSet);
+
+        LineSet dataset = new LineSet(xAxisLabels, sentValues);
+        dataset.setColor(parseColor("#b01cff"))
+                .setDotsColor(parseColor("#1cb7ff"))
+                .setDashed(new float[]{15f, 10f})
+                .setThickness(6)
+                .beginAt(0);
+        lineGraph.addData(dataset);
+    }
+
+    private void setGraphAttributes() {
+
+        //todo REVIEW I forgot why I did this but I believe it crashes if we don't set this value to 100
+        setHighestValueTo100();
```

Comment:
If this crashes, then your previous setHighestValue function did something wrong...